### DISCUSSION TITLE
How to check whether a string is a template?

### DISCUSSION THREAD

**Discussion by sisp** [pallets/jinja#1736] [Q&A]
What is the best way to check whether a string is a template, i.e. whether it contains template syntax? I've looked into [`Environment.lex(...)`](https://github.com/pallets/jinja/blob/5d9ece6d65ae4c2f57dd2bdccc28943984a767f9/src/jinja2/environment.py#L619-L638) to get the tokens and derive the check from that information by checking whether there is any token of type other than `data` in the [tokens](https://github.com/pallets/jinja/blob/5d9ece6d65ae4c2f57dd2bdccc28943984a767f9/src/jinja2/lexer.py#L61-L109) list:

```python
def is_template_string(value: str, env: jinja2.Environment) -> bool:
    return any(token_type != jinja2.lexer.TOKEN_DATA for _, token_type, _ in env.lex(value))
```

Is this approach correct and is there anything easier I could do? And would it be an interesting feature to be added to the library?

**Comment by creatorrr** [comment]
this might be simpler and faster, no?

```python
def is_jinja(s):
    return re.search('{{.+}}',s) or re.search('{%.+%}',s)
```

**Reply by creatorrr** [reply]
from: https://stackoverflow.com/questions/66041586/check-whether-string-is-jinja2-template-or-not
